src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Suppressor.cs

```diff
@@ -51,9 +51,10 @@ public override void ReportSuppressions(SuppressionAnalysisContext context)
                     bool shouldSuppressDiagnostic =
                         location.SourceTree is SyntaxTree sourceTree &&
                         sourceTree.GetRoot().FindNode(location.SourceSpan) is SyntaxNode syntaxNode &&
-                        BinderInvocation.IsCandidateSyntaxNode(syntaxNode) &&
+                        (syntaxNode as InvocationExpressionSyntax ?? syntaxNode.Parent as InvocationExpressionSyntax) is InvocationExpressionSyntax invocation &&
```

Comment:
>syntaxNode.Parent as InvocationExpressionSyntax [](http://example.com/codeflow?start=69&length=47)

Could you please elaborate with example more on checking the parent which we were not checking before?

Reply:
It wasn't previously required to check MemberAccessExpressions for warnings because the diagnostic was always on the entire InvocationExpressionsSyntax, but with the changes to the trim analyzer, the warnings will be on the MemberAccessExpression instead.

```csharp
// IL2026 location before
{|MethodWithRuc(arg1, arg1)|};
// IL2026 location after
{|MethodWithRuc|}(arg1, arg1);
```

Reply:
Thanks!

Can you please add comment with this example?

Reply:
Sure thing, added a comment